packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -29,13 +29,20 @@ internal sealed class MrwSerializationTypeProvider : TypeProvider
         private const string JsonModelWriteCoreMethodName = "JsonModelWriteCore";
         private const string WriteAction = "writing";
         private const string ReadAction = "reading";
+        private const string AdditionalRawDataVarName = "serializedAdditionalRawData";
         private readonly ParameterProvider _utf8JsonWriterParameter = new("writer", $"The JSON writer.", typeof(Utf8JsonWriter));
         private readonly ParameterProvider _serializationOptionsParameter =
             new("options", $"The client options for reading and writing models.", typeof(ModelReaderWriterOptions));
+        private readonly ParameterProvider _deserializationOptionsParameter =
```

Comment:
shouldn't need the nullable version so you can reuse the _serializationOptionsParameter